apollo-client/src/main/java/com/ctrip/framework/apollo/internals/DefaultConfig.java

```diff
@@ -103,151 +56,28 @@ public String getProperty(String key, String defaultValue) {
-  private void updateConfig(Properties newConfigProperties, ConfigSourceType sourceType) {
-    m_configProperties.set(newConfigProperties);
-    m_sourceType = sourceType;
-  }
-
-  private Map<String, ConfigChange> updateAndCalcConfigChanges(Properties newConfigProperties,
-      ConfigSourceType sourceType) {
-    List<ConfigChange> configChanges =
-        calcPropertyChanges(m_namespace, m_configProperties.get(), newConfigProperties);
-
-    ImmutableMap.Builder<String, ConfigChange> actualChanges =
-        new ImmutableMap.Builder<>();
-
-    /** === Double check since DefaultConfig has multiple config sources ==== **/
-
-    //1. use getProperty to update configChanges's old value
-    for (ConfigChange change : configChanges) {
-      change.setOldValue(this.getProperty(change.getPropertyName(), change.getOldValue()));
-    }
-
-    //2. update m_configProperties
-    updateConfig(newConfigProperties, sourceType);
-    clearConfigCache();
-
-    //3. use getProperty to update configChange's new value and calc the final changes
-    for (ConfigChange change : configChanges) {
-      change.setNewValue(this.getProperty(change.getPropertyName(), change.getNewValue()));
-      switch (change.getChangeType()) {
-        case ADDED:
-          if (Objects.equals(change.getOldValue(), change.getNewValue())) {
-            break;
-          }
-          if (change.getOldValue() != null) {
-            change.setChangeType(PropertyChangeType.MODIFIED);
-          }
-          actualChanges.put(change.getPropertyName(), change);
-          break;
-        case MODIFIED:
-          if (!Objects.equals(change.getOldValue(), change.getNewValue())) {
-            actualChanges.put(change.getPropertyName(), change);
-          }
-          break;
-        case DELETED:
-          if (Objects.equals(change.getOldValue(), change.getNewValue())) {
-            break;
-          }
-          if (change.getNewValue() != null) {
-            change.setChangeType(PropertyChangeType.MODIFIED);
-          }
-          actualChanges.put(change.getPropertyName(), change);
-          break;
-        default:
-          //do nothing
-          break;
-      }
-    }
-    return actualChanges.build();
-  }
-
-  private Properties loadFromResource(String namespace) {
-    String name = String.format("META-INF/config/%s.properties", namespace);
-    InputStream in = ClassLoaderUtil.getLoader().getResourceAsStream(name);
-    Properties properties = null;
-
-    if (in != null) {
-      properties = propertiesFactory.getPropertiesInstance();
-
-      try {
-        properties.load(in);
-      } catch (IOException ex) {
-        Tracer.logError(ex);
-        logger.error("Load resource config for namespace {} failed", namespace, ex);
-      } finally {
-        try {
-          in.close();
-        } catch (IOException ex) {
-          // ignore
-        }
-      }
-    }
-
-    return properties;
+    Set<String> fromRepository = this.getPropertyNamesFromRepository();
```

Comment:
How about we only include propertyNamesFromRepository and propertyNamesFromAdditional? I have a little concern that if system property and system env are included, that might cause some compatibility issues, though this looks like the correct implementation.

Reply:
resolved